### PR TITLE
Feat/recording indicator

### PR DESCRIPTION
**Pull request by OCannings**

### CODE REVIEW

File: apps/juicebox_web/assets/js/reactions/styles.scss [hunk 1]
```diff
@@ -13,3 +13,46 @@
...
+      background-color: red;
+      width: 0;
+      height: 2px;
+      transition: width 2.5s;
+      transition-timing-function: linear;
+    }
+
+    &.recording {
+      &::after {
+        visibility: visible;
```
Comment (ljkbennett) [inline apps/juicebox_web/assets/js/reactions/styles.scss:44]: Properties should be ordered width, visibility

[Read more about it here.](https://github.com/brigade/scss-lint/blob/master/lib/scss_lint/linter/README.md#propertysortorder)

File: apps/juicebox_web/assets/js/reactions/styles.scss [hunk 2]
```diff
@@ -13,3 +13,46 @@
...
+      video {
+        display: none;
+      }
+    }
+
+    &::after {
+      content: ' ';
+      display: block;
+      visibility: hidden;
+      background-color: red;
```
Comment (ljkbennett) [inline apps/juicebox_web/assets/js/reactions/styles.scss:35]: Color `red` should be written in hexadecimal form as `#ff0000`

[Read more about it here.](https://github.com/brigade/scss-lint/blob/master/lib/scss_lint/linter/README.md#colorkeyword)

File: apps/juicebox_web/assets/js/reactions/styles.scss [hunk 3]
```diff
@@ -13,3 +13,46 @@
...
+    &.offline {
+      video {
+        display: none;
+      }
+    }
+
+    &::after {
+      content: ' ';
+      display: block;
+      visibility: hidden;
```
Comment (ljkbennett) [inline apps/juicebox_web/assets/js/reactions/styles.scss:34]: Properties should be ordered content, display, width, height, visibility, background-color, transition, transition-timing-function

[Read more about it here.](https://github.com/brigade/scss-lint/blob/master/lib/scss_lint/linter/README.md#propertysortorder)

File: apps/juicebox_web/assets/js/reactions/recorder.js [hunk 4]
```diff
@@ -12,86 +13,110 @@ class ReactionRecorder extends Component {
...
-      webcamConnected: false,
-      recording: null,
+      isConnected: false,
+      isRecording: false,
+      video: null,
+      stream: null,
     }
   }
 
+  startButton = () =>
```
Comment (ljkbennett) [inline apps/juicebox_web/assets/js/reactions/recorder.js:23]: Parsing error: Unexpected token =

**Comment by ljkbennett** [comment]
Ebert has finished reviewing this Pull Request and has found:

* 3 possible new issues (including those that may have been commented here).

You can see more details about this review at https://ebertapp.io/github/ItsJuice/juicebox/pulls/64.

File: apps/juicebox_web/assets/js/reactions/recorder.js [hunk 5]
```diff
@@ -12,86 +13,110 @@ class ReactionRecorder extends Component {
...
+      <div className="reaction-recorder">
+        <div className={`webcam
+          ${!isConnected ? 'offline' : ''}
+          ${isRecording ? 'recording' : ''}`}>
+
+          { show(this.stopButton).when(isConnected) }
+          <video src={video || stream} loop autoPlay></video>
+
+          <div className="recorder-controls">
+            { show(this.startButton).when(!isConnected) }
```
Comment (ljkbennett) [inline apps/juicebox_web/assets/js/reactions/recorder.js:57]: I like it. Only thing is the naming of the `isConnected` function as I was trying to work out how it called the function until I realised it was also the name of the state variable.

**Review by ljkbennett — APPROVED** [review]

File: apps/juicebox_web/assets/js/reactions/styles.scss (hunk 2 above)
Comment (ghost) [reply apps/juicebox_web/assets/js/reactions/styles.scss:35]: Why is Herbert critising this I wonder?

Comment (OCannings) [reply apps/juicebox_web/assets/js/reactions/styles.scss:35]: The link explains it: "Color keywords look like variables but are not variables." which is fair enough, I suppose.

File: apps/juicebox_web/assets/js/reactions/recorder.js (hunk 5 above)
Comment (OCannings) [reply apps/juicebox_web/assets/js/reactions/recorder.js:57]: Good point, especially considering I was going to allow `when` to accept a predicate function!

**Pull request merged**